### PR TITLE
Fix for Errno::ENAMETOOLONG in Thumbnails

### PR DESCRIPTION
**Pull request by codeodor**
* also includes a test for TempfileFactory to be sure it does not revert to similar behavior

### CODE REVIEW

**Comment by codeodor** [comment]
Fixed hound issues and rebased.

**Comment by codeodor** [comment]
What do you think of using Paperclip's Tempfile here instead of Ruby's?

**Comment by codeodor** [comment]
If this looks acceptable, I'll rebase it into a single commit. Let me know. Thanks!

**Comment by jyurek** [comment]
Get your head in the game! I've seen a video of a guy soloing the whole raid in 12 minutes!

;)

Sammy Larbi wrote:

File: lib/paperclip/thumbnail.rb [hunk 1]
```diff
@@ -63,8 +63,8 @@ def convert_options?
     # that contains the new image.
     def make
       src = @file
-      dst = Tempfile.new([@basename, @format ? ".#{@format}" : ''])
-      dst.binmode
+      filename = [@basename, @format ? ".#{@format}" : ""].join
+      dst = TempfileFactory.new.generate(filename)
```
Comment (jyurek) [inline lib/paperclip/thumbnail.rb]: Do we not still need `binmode`?

**Comment by jyurek** [comment]
Oh, duh, `TempfileFactory` handles that. Yeah, this looks good to me, if you could rebase, that would be awesome.

File: lib/paperclip/thumbnail.rb (hunk 1 above)
Comment (jyurek) [inline lib/paperclip/thumbnail.rb]: Oh, duh. `TempfileFactory` handles that.

**Comment by codeodor** [comment]
Ok, done!

**Comment by jyurek** [comment]
Thanks!

**Comment by codeodor** [comment]
My pleasure :+1: and thanks for Paperclip! :trophy:

**Pull request merged**